pandasai/pipelines/chat/generate_chat_pipeline.py

```diff
@@ -62,7 +62,7 @@ def __init__(
                 ),
                 CachePopulation(skip_if=self.is_cached),
                 CodeExecution(
-                    before_execution=on_code_execution,
```

Comment:
In pipeline, we check for that, therefore if it's none, nothing is executed:

```
# Callback function before execution
if logic.before_execution is not None:
    logic.before_execution(data)
```

Reply:
Check out the fix: https://github.com/Sinaptik-AI/pandas-ai/commit/3b4ef7b89ddc50bbd923c27ae00087e71ef3a0e3

Reply:
@gventuri This doesn't solve the issue the last_code_executed has:
{
    "type": "string",
    "value": f"The employee with the highest salary is {employee_name_highest_salary} with a salary of ${highest_salary}
    
  Reason is before_execution

Reply:
Agree with @ArslanSaleem . I fixed the issue by setting the callback as `on_execution` instead of `before_execution` (see #1015 ). The callback can't access the executed code before `CodeExecution.execute`, since the executed code is only generated during `CodeExecution.execute`.